pkg/kapp/diff/filter.go

```diff
@@ -0,0 +1,233 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package diff
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/k14s/kapp/pkg/kapp/matcher"
+	res "github.com/k14s/kapp/pkg/kapp/resources"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+type ChangeSetFilter struct {
+	df     DiffFilter
+	Filter string
+}
+
+type DiffFilter struct {
+	Kinds          []string
+	Namespaces     []string
+	Names          []string
+	KindNames      []string
+	KindNamespaces []string
+	KindNsNames    []string
+	LabelSelector  []string
+	BoolFilter     *BoolFilter
+}
+
+type BoolFilter struct {
```

Comment:
lets also add `ops []string` to allow selecting changes by their operation.

Reply:
I have created a `type OpsFilter []ChangeOp` will it be fine ?